yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -540,41 +427,28 @@ export class L2Block {
       return layers[layers.length - 1][0];
     };
 
-    const leafCount = this.newCommitments.length / MAX_NEW_COMMITMENTS_PER_TX;
     const leafs: Buffer[] = [];
 
-    for (let i = 0; i < leafCount; i++) {
-      const commitmentsPerBase = MAX_NEW_COMMITMENTS_PER_TX;
-      const nullifiersPerBase = MAX_NEW_NULLIFIERS_PER_TX;
-      const publicDataUpdateRequestsPerBase = MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX;
-      const l2ToL1MsgsPerBase = MAX_NEW_L2_TO_L1_MSGS_PER_TX;
-      const commitmentsBuffer = Buffer.concat(
-        this.newCommitments.slice(i * commitmentsPerBase, (i + 1) * commitmentsPerBase).map(x => x.toBuffer()),
-      );
-      const nullifiersBuffer = Buffer.concat(
-        this.newNullifiers.slice(i * nullifiersPerBase, (i + 1) * nullifiersPerBase).map(x => x.toBuffer()),
-      );
-      const publicDataUpdateRequestsBuffer = Buffer.concat(
-        this.newPublicDataWrites
-          .slice(i * publicDataUpdateRequestsPerBase, (i + 1) * publicDataUpdateRequestsPerBase)
-          .map(x => x.toBuffer()),
-      );
-      const newL2ToL1MsgsBuffer = Buffer.concat(
-        this.newL2ToL1Msgs.slice(i * l2ToL1MsgsPerBase, (i + 1) * l2ToL1MsgsPerBase).map(x => x.toBuffer()),
-      );
-      const encryptedLogsHashKernel0 = L2Block.computeKernelLogsHash(this.newEncryptedLogs.txLogs[i]);
+    for (let i = 0; i < this.body.txEffects.length; i++) {
+      const txEffect = this.body.txEffects[i];
 
-      const unencryptedLogsHashKernel0 = L2Block.computeKernelLogsHash(this.newUnencryptedLogs.txLogs[i]);
+      const commitmentsBuffer = Buffer.concat(txEffect.newNoteHashes.map(x => x.toBuffer()));
+      const nullifiersBuffer = Buffer.concat(txEffect.newNullifiers.map(x => x.toBuffer()));
+      const publicDataUpdateRequestsBuffer = Buffer.concat(txEffect.newPublicDataWrites.map(x => x.toBuffer()));
+      const newL2ToL1MsgsBuffer = Buffer.concat(txEffect.newL2ToL1Msgs.map(x => x.toBuffer()));
+      const encryptedLogsHashKernel0 = L2Block.computeKernelLogsHash(txEffect.logs!.encryptedLogs);
+      const unencryptedLogsHashKernel0 = L2Block.computeKernelLogsHash(txEffect.logs!.unencryptedLogs);
 
       const inputValue = Buffer.concat([
```

Comment:
See comment on `tx_effect` later. This should almost entirely be replaced with `leafs.push(txEffect.hash());`